### ISSUE TITLE
Allow users to set encoding of files manually

### ISSUE THREAD

**Issue by yxxthu** [spyder-ide/spyder#6554]
## Description
I have an interface that has to use gbk encoding, however, when I change the file encoding to gbk, it will use ascii or latin-1-gussed. I am really confused, the code works on vs code or terminal, but it doesn't work on spyder
**What steps will reproduce the problem?**

1. write a file and save it in gbk encoding
2. open it in spyder, and the encoding change to ascii or latin-1-guessed
3. also you can see some random charaters,like ÕæÂé·³

**What is the expected output? What do you see instead?**

**Please provide any additional information below**

## Version and main components

* Spyder Version: 3.2.4 
* Python Version: 3.6.3
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
numpy >=1.7      :  1.13.3 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
rope >=0.9.4     :  0.10.5 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
cython >=0.21    :  0.26.1 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
IPython >=4.0    :  6.1.0 (OK)
pylint >=0.25    :  1.7.4 (OK)

```

**Comment by ccordoba12** [comment]
We use the Chardet library to detect not so common encodings, and it seems Chardet is not able to detect gbk:

https://github.com/chardet/chardet/issues/94

Unfortunately, we can't do anything about this, sorry.

**Comment by yxxthu** [comment]
there are also other guys find the problem, 
http://www.itkeyword.com/doc/4161783280580393638/can-the-editor-encoding-setting-in-spyder-be-changed

I think there must be something wrong of spyder sourcecode, it auto-detect the coding and do not allow the users to change it, it's really annoying when the auto-detect result is wrong

**Comment by ccordoba12** [comment]
I don't think we will be able to address in a year or more, but we'll try to get to it.

**Comment by yxxthu** [comment]
Thanks a lot!

**Comment by CAM-Gerlach** [comment]
Closed as duplicate of #5551 .
